org.adempiere.base.process/src/org/compiere/process/InOutGenerate.java

```diff
@@ -173,7 +173,7 @@ protected String doIt () throws Exception
 			if (p_C_BPartner_ID != 0)
 				m_sql.append(" AND o.C_BPartner_ID=?");					//	#3
 		}
-		m_sql.append(" ORDER BY M_Warehouse_ID, PriorityRule, M_Shipper_ID, C_BPartner_ID, C_BPartner_Location_ID, C_Order_ID");
+		m_sql.append(" ORDER BY M_Warehouse_ID, PriorityRule, M_Shipper_ID, DateOrdered, C_BPartner_ID, C_BPartner_Location_ID, C_Order_ID");
```

Comment:
this change is not explain in the ticket. if it is for a separate issue, create another ticket for it.

Reply:
you are right, it was another patch. @hengsin, what do you think: will it be a generally accepted rule to take orders with an earlier date a higher priority than the ordering by the bpartner ID? Then we'll create another ticket.